envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/asssertions/EnvoyAssertions.kt

```diff
@@ -0,0 +1,81 @@
+package pl.allegro.tech.servicemesh.envoycontrol.asssertions
+
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.ObjectAssert
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoy.EnvoyContainer
+
+class EnvoyAssertions(
+    val protocol: String,
+    val path: String,
+    val method: String,
+    val clientName: String,
+    val clientIp: String
+)
+
+fun ObjectAssert<EnvoyContainer>.hasNoRBACDenials() = satisfies {
+    val admin = it.admin()
+    assertThat(admin.statValue("http.ingress_http.rbac.denied")?.toInt()).isZero()
+    assertThat(admin.statValue("http.ingress_https.rbac.denied")?.toInt()).isZero()
+    assertThat(admin.statValue("http.ingress_http.rbac.shadow_denied")?.toInt()).isZero()
+    assertThat(admin.statValue("http.ingress_https.rbac.shadow_denied")?.toInt()).isZero()
+
+    assertThat(it.logRecorder.getRecordedLogs()).isEmpty()
```

Comment:
Right, this doesn't make much sense now, because the crucial part is in complete change only: https://github.com/allegro/envoy-control/compare/master...shadowruleslogging#diff-97cd24ee896a0ebbbf389ba5d4b434ecR689
I will try to refactor it so this PR alone will make more sense.

Reply:
Comment is enough.

Reply:
refactored and fixed